File: Data Structures & Algorithms/max-water-container/submission-0.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        max_circumeference :int = 0 ; length : int = len(heights)
        right : int = 0  ; left : int = len(heights) - 1

        while right < left :
             if heights[right] > heights[left] : 
                curr_cirumference : int = heights[left]*(left-right)
                left -=1
             else:
                 curr_cirumference : int = heights[right]*(left-right)
                 right +=1
             if  curr_cirumference > max_circumeference : max_circumeference = curr_cirumference

        return max_circumeference

File: Data Structures & Algorithms/binary-tree-preorder-traversal/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> preorderTraversal(TreeNode root) {
        //empty is empty ??
        if(root == null) return new ArrayList<>();
        List<Integer> L = new ArrayList<>();
        // We construct Our preorder Traversal List ...
        preorderAddition(L,root);

        return L ; 
        
    }

    private void preorderAddition(List<Integer> L ,TreeNode root)
    {
        if(root == null) return  ; 
        // Addition the root value
        L.add(root.val);
        //Then recursively we add the left nodes values first 
        preorderAddition(L,root.left);
        // and then the right ones afterwards
        preorderAddition(L,root.right);
        

    }
}